src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/token/Token.java

```diff
@@ -0,0 +1,163 @@
+package de.terrestris.shogun2.model.token;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+import org.hibernate.annotations.Type;
+import org.joda.time.DateTime;
+import org.joda.time.ReadableDateTime;
+
+import de.terrestris.shogun2.model.PersistentObject;
+import de.terrestris.shogun2.model.User;
+
+/**
+ * Base class for all tokens.
+ *
+ * @author Daniel Koch
+ *
+ */
+@Entity
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
+public abstract class Token extends PersistentObject {
+
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 1L;
+
+	/**
+	 * The token validity period (in hours).
+	 *
+	 * The Default is to 24.
+	 *
+	 */
+	private static final int expiration = 24;
+
+	/**
+	 * The token string itself.
+	 */
+	private String token;
+
+	/**
+	 * The expiration date of the token. Will be set
+	 */
+	@Column(updatable = false)
+	@Type(type = "org.jadira.usertype.dateandtime.joda.PersistentDateTime")
+	private final ReadableDateTime expirationDate;
+
+	/**
+	 * The user who has requested the token. Hereby one user can have one
+	 * token and one token can be used by one user (at the same time) only.
+	 */
+	@OneToOne
+	@JoinColumn(name = "USER_ID")
+	private User user;
+
+	/**
+	 * Constructor
+	 */
+	protected Token(int expiration) {
+		// set the expiration date
+		this.expirationDate = DateTime.now().plusHours(expiration);
```

Comment:
Ahh I see, because the childclass wants to call it with its class constant, right? Maybe @buhner can say sth. wrt this.